### PR TITLE
Support for Rails 3.1 asset pipeline

### PR DESCRIPTION
**Pull request by jweslley**
Since Rails 3.1 is almost out, I guess web-app-theme can be improved when using the new Rails feature, the asset pipeline. In addition to explore this new feature, changing the application theme can be easy than ever, just editing a single line in `application.css` file*. Thus, the `application.css` could be so simple as the following:

```
/*
 *= require 'web-app-theme'
 *= require 'web-app-theme/default'
 *= require_tree . 
 */
```

The first line include the web-app-theme's basic/common styles previously located in `base.css` file. While the second one includes all theme-specific styles.
- Yes, nowadays we can use the theme generator passing `--no-layout` flag, but this command generates a bunch of files (css/images/fonts).

### CODE REVIEW

**Comment by nathanvda** [comment]
Nice work. My approach was more conservative :)

File: app/assets/stylesheets/web-app-theme/default.css.erb [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
+  src: url(<%= asset_path 'web-app-theme/default/museo700-regular-webfont.eot' %>);
+  src: local("☺"), url(<%= asset_path 'web-app-theme/default/museo700-regular-webfont.woff' %>) format("woff"), url(<%= asset_path 'web-app-theme/default/museo700-regular-webfont.ttf' %>) format("truetype"), url(<%= asset_path 'web-app-theme/default/museo700-regular-webfont.svg#webfontOC5Ke5xr' %>) format("svg");
   font-weight: normal;
   font-style: normal;
 }
 
 body {
   color: #111111;
-  background: #c4c4c4 url("images/bgd.jpg");
+  background: #c4c4c4 url(<%= asset_path 'web-app-theme/default/bgd.png' %>);
```
Comment (rgabo) [inline app/assets/stylesheets/web-app-theme/default.css.erb]: @rgabo ♥ png, but we should stick to what we have ;) (it's a JPG)

Comment (jweslley) [inline app/assets/stylesheets/web-app-theme/default.css.erb]: Thanks @rgabo.

**Comment by rgabo** [comment]
any update on the rails 3.1-compat pull requests, @pilu? We're using this pull request. We ran into an issue with asset:precompile which is a known issue with Rails 3.1.0 official, also our configuration was generated with rc6 and some of the asset-related settings were misconfigured (cache, digest, etc). Otherwise, this pull request worked great. No wonder it is issue #42 :)

**Comment by leehambley** [comment]
Just spotted this, looks like awesome work, can't wait to see where it goes next!

**Comment by dryaf** [comment]
nice! i hope it s getting pulled soon

**Comment by gravityblast** [comment]
hi guys, I already merged something from other forks. Pull from my master branch, check if everything is ok, and then I'll merge in your changes. Thank you!

File: app/assets/stylesheets/web-app-theme/default.css.erb (hunk 1 above)
Comment (NARKOZ) [inline app/assets/stylesheets/web-app-theme/default.css.erb]: @jweslley please update it with asset helpers. no need for erb.

``` sass
image-url('web-app-theme/default/bgd.png')
```

**Comment by willthames** [comment]
I've attempted to merge pilu/master back into jweslley/master. 
git@github.com:willthames/web-app-theme.git

Looking at the network graph it doesn't seem to reflect how the branches do come together, however - I may have made a git mistake somewhere (perhaps I should have merged the other way around). 

I'm not clear how assets are generated now, since the web-app-theme/assets generator no longer exists. 

Otherwise my fork reflects most of pilu/master's recent history and works with 3.2 as per slainer68's fix in #49.

**Comment by gravityblast** [comment]
hi guys, I can't automatically merge this request, maybe some conflicts. if you try to rebase or fix that I'll merge it asap, thank you!

**Comment by RobertAudi** [comment]
Please merge one of the asset pipeline compatible pull requests so that we can use web-app-theme with rails 3.2...

**Comment by gravityblast** [comment]
Hi, if you guys can manage this now I can give you write access to the repository. Let me know if you can, thank you!

**Comment by sheam** [comment]
Is there a way I can use this pull in my rails 3.2 project? What line do I need in my gem file?

**Comment by rgabo** [comment]
Is there a way you can close this PR? It has been showing up in my PR list since 2012 😉 Happy 2019, everyone!

**Comment by elia** [comment]
@rgabo does unsubscribing do the trick? Happy 2019! 😜

**Comment by rgabo** [comment]
@elia it doesn't 😢 

it_doesnt

**Comment by elia** [comment]
@rgabo I'll close this, happy github inbox-zero! 😉

**Comment by rgabo** [comment]
Grazie mille, @elia! 😄

**Pull request closed without merging**